SPECS/selinux-policy/0038-enable-liveos-iso-flow.patch

```diff
@@ -0,0 +1,223 @@
+
+The problem is that selinux denies many of these processes access
+to a number of resources (mostly file descriptors, but here is also
+access to the boot media, and other resources).
+
+This patch allows these failing processes access to the resources
+they need.
+---
+ policy/modules/admin/usermanage.te   |  2 ++
+ policy/modules/apps/loadkeys.te      |  2 ++
+ policy/modules/kernel/devices.if     | 18 ++++++++++++++++++
+ policy/modules/kernel/kernel.if      | 19 +++++++++++++++++++
+ policy/modules/kernel/kernel.te      |  1 +
+ policy/modules/system/authlogin.te   |  1 +
+ policy/modules/system/getty.te       |  2 ++
+ policy/modules/system/init.te        |  7 +++++++
+ policy/modules/system/locallogin.te  |  1 +
+ policy/modules/system/selinuxutil.te |  2 ++
+ policy/modules/system/udev.te        |  2 ++
+ 11 files changed, 57 insertions(+)
+
+diff --git a/policy/modules/admin/usermanage.te b/policy/modules/admin/usermanage.te
+index 57a9848bc..ef24fb85b 100644
+--- a/policy/modules/admin/usermanage.te
++++ b/policy/modules/admin/usermanage.te
+@@ -317,6 +317,7 @@ allow passwd_t self:shm create_shm_perms;
+ allow passwd_t self:sem create_sem_perms;
+ allow passwd_t self:msgq create_msgq_perms;
+ allow passwd_t self:msg { send receive };
++allow passwd_t kernel_t:fd use;
+ 
+ allow passwd_t crack_db_t:dir list_dir_perms;
+ read_lnk_files_pattern(passwd_t, crack_db_t, crack_db_t)
+@@ -492,6 +493,7 @@ allow useradd_t self:unix_dgram_socket create_socket_perms;
+ allow useradd_t self:unix_stream_socket create_stream_socket_perms;
+ allow useradd_t self:unix_dgram_socket sendto;
+ allow useradd_t self:unix_stream_socket connectto;
++allow useradd_t kernel_t:fd use;
+ 
+ # for getting the number of groups
+ kernel_read_kernel_sysctls(useradd_t)
+diff --git a/policy/modules/apps/loadkeys.te b/policy/modules/apps/loadkeys.te
+index 2263e9dba..61d098341 100644
+--- a/policy/modules/apps/loadkeys.te
++++ b/policy/modules/apps/loadkeys.te
+@@ -21,6 +21,8 @@ allow loadkeys_t self:capability { dac_override dac_read_search setuid sys_tty_c
+ allow loadkeys_t self:fifo_file rw_fifo_file_perms;
+ allow loadkeys_t self:unix_stream_socket { connect create };
+ 
++kernel_getattr_unlabeled_dirs(loadkeys_t)
++kernel_use_fds(loadkeys_t)
+ kernel_read_system_state(loadkeys_t)
+ 
+ init_use_fds(loadkeys_t)
+diff --git a/policy/modules/kernel/devices.if b/policy/modules/kernel/devices.if
+index 35e14a021..68c9b8720 100644
+--- a/policy/modules/kernel/devices.if
++++ b/policy/modules/kernel/devices.if
+@@ -2206,6 +2206,24 @@ interface(`dev_setattr_input_dev',`
+ 	allow $1 event_device_t:chr_file setattr;
+ ')
+ 
++########################################
++## <summary>
++##	Get the attributes of the event devices.
++## </summary>
++## <param name="domain">
++##	<summary>
++##	Domain allowed access.
++##	</summary>
++## </param>
++#
++interface(`dev_read_input_dev',`
++	gen_require(`
++		type event_device_t;
++	')
++
++	allow $1 event_device_t:chr_file read;
```

Comment:
I would change the `read` to `read_chr_file_perms` so you get useful related perms like `getattr` (`stat()`). The extra permissions aren't concerning, and we won't need to change the policy if that pops up.

Reply:
updated - thx Chris.